### ISSUE TITLE
[a11y]: MultiSelect [default] - selected options bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15256]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.39.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.39.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
The selected options names are not announced by JAWS

Once the user makes the selection, the selected number of options is announced by JAWS. However, the names of those are not announced. Hence the user will not know which are they. Note: Working fine in iPhone VO. This issue is applicable for all the variants.

### Recommended Fix
Either the names of the selected options should also be identified near the control with the total number of selection. Or while navigating through the options within the selectbox SR should announce the selected state of the currently selected options. 

When the state of an element i.e. disabled, selected, expanded, currently active etc. is not conveyed for the screen reader user, it can be misleading for the user. . this impacts the user interaction.

### WCAG 2.1 Violation
4.1.2 Name, Role, Value
